### PR TITLE
Group medicine administration by 4 hours + Support for archiving medicines + Administration Activity View + migrate `useDispatch` to `useQuery`

### LINKED ISSUE #6184: Support for archiving Medicine Administration Record
Users may enter incorrect records for MAR. 

Hence, users should be able to archive an existing MAR and create a new MAR for the same instead.

### LINKED ISSUE #6341: Medicine Administrations to be grouped based on common frequency/interval instead of every hour
- There is so much space wasted in the MAR table

image

cc: @aparnacoronasafe

### LINKED ISSUE #6442: 🛠️ Replace `useDispatch` w. `useQuery`/`request`: Medicine (`src/Components/Medicine/**`)

### LINKED ISSUE #6459: Administer Medicine Slider and Modal not responsive
**Describe the bug**
Text is going outside Modal in mobile view and input box size is very small in desktop view for slider.

**To Reproduce**

1. Go to https://care.coronasafe.in/facility/657c32be-d584-476c-9ce2-0412f0e7692e/patient/80e47d10-9ed8-4193-ade6-d69aa45999e9/consultation/b81e5e1f-fd11-4b38-aa51-14bea5583c76/medicines
2. Click on `Administer`
3. Open in mobile view
5. Now click on `Administer Medicines`
6. See error

**Screenshots**
image
image

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by rithviknishad**
### <samp>🤖 Generated by Copilot at 1f75446</samp>

Improved the usability and responsiveness of the `PrescriptionAdministrationsTable` component by changing the pagination logic and layout to show full days of data.

## Required Backends

- https://github.com/coronasafe/care/pull/1630

## Proposed Changes

- Fixes #6341
- Fixes #6442 
- Fixes #6184
- Fixes #6459

image

image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at 1f75446</samp>

* Import `useBreakpoints` hook to determine the number of days per page based on screen size ([link](https://github.com/coronasafe/care_fe/pull/6396/files?diff=unified&w=0#diff-72e4733fee5da36cd347a741a637541d37363b7ecd2280dcc5ce01d6aa173150R24))
* Declare `daysPerPage` variable using `useBreakpoints` hook with default value of 1 and higher values for larger screens ([link](https://github.com/coronasafe/care_fe/pull/6396/files?diff=unified&w=0#diff-72e4733fee5da36cd347a741a637541d37363b7ecd2280dcc5ce01d6aa173150R52-R56))
* Update `usePagination` hook to use `daysPerPage` variable for `perPage` and `slots` properties, which represent the pagination range and the table columns respectively ([link](https://github.com/coronasafe/care_fe/pull/6396/files?diff=unified&w=0#diff-72e4733fee5da36cd347a741a637541d37363b7ecd2280dcc5ce01d6aa173150L56-R63))
* Modify `getAdministrationBounds` function to round the pagination range to whole days instead of hours, to align with `daysPerPage` variable ([link](https://github.com/coronasafe/care_fe/pull/6396/files?diff=unified&w=0#diff-72e4733fee5da36cd347a741a637541d37363b7ecd2280dcc5ce01d6aa173150L603-R614))

### CODE REVIEW

File: src/Components/Notifications/NotificationsList.tsx
Comment (rithviknishad) [inline src/Components/Notifications/NotificationsList.tsx:1]: This file just got prettified while rebasing with the latest `develop` branch automatically.

**Comment by nihal467** [comment]
@rithviknishad 
image

- [ ] when we click on the administration tick mark, tooltip is clipping out
- [ ] once we click on it, the page is shuttering 

image

in mobile view its completely breaking

**Comment by nihal467** [comment]
@rithviknishad 
image

- [ ] it is stretching the page in the mobile view
- [ ] the data are over lapping in mobile view

image

- [ ] in the normal view, the data are overlapping

**Comment by rithviknishad** [comment]
The stretch issue of the page cannot be solved easily, if I make the table horizontally scrollable, it'll clip out the Popover too (all vertical content) too. This issue is a limitation by the CSS itself. https://stackoverflow.com/a/6433475/7887936 

`overflow-x-auto` causes vertical contents too to be clipped/scrollable instead of making it visible (that'll include the Popover overflow too)
`overflow-x-auto overflow-y-visible` has no effect.

cc: @Ashesh3 @khavinshankar @sainak 

We could make this in a dedicated page (for mobile view alone) maybe in a later PR?

**Comment by rithviknishad** [comment]
@nihal467 the other mentioned issues has been fixed.

image

**Comment by rithviknishad** [comment]
The responsive design needs to be further discussed

**Comment by nihal467** [comment]
@rithviknishad 
image

- the archive button seem to be misaligned when the medicine is created data is shown
- max and min dose UI is breaking for PRN prescription

image

- unable to discontinue a medicine

image

- tooltip is overlapping

**Comment by nihal467** [comment]
@rithviknishad 
image

add a medicine and discontinue it, the count in the hidden prescription wont get auto updated, even though the medicine is hidden now, unless we do a manual reload of page

image

in mobile view the prescription data seems to be overflowing outside the padding

image

hide archive button for the discontinued medicine

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**